mobile/src/main/java/ch/epfl/sdp/mobile/ui/features/Navigation.kt

```diff
@@ -0,0 +1,64 @@
+package ch.epfl.sdp.mobile.ui.features
+
+import androidx.compose.animation.Crossfade
+import androidx.compose.runtime.*
+import ch.epfl.sdp.mobile.data.api.AuthenticationApi.User
+import ch.epfl.sdp.mobile.ui.LocalAuthenticationApi
+import ch.epfl.sdp.mobile.ui.features.authentication.StatefulAuthenticationScreen
+import ch.epfl.sdp.mobile.ui.features.home.StatefulHome
+import kotlinx.coroutines.flow.map
+
+/**
+ * The root of the application, which determines the top-level state of the navigation. More
+ * precisely, it takes care of displaying the proper screens if the user is connected or
+ * disconnected.
+ */
+@Composable
+fun Navigation() {
+  val state by rememberUserState()
+  Crossfade(state) {
+    when (it) {
+      is UserState.Authenticated -> StatefulHome(it.user)
+      UserState.NotAuthenticated -> StatefulAuthenticationScreen()
+      UserState.Loading -> Unit
+    }
+  }
+}
+
+/** An interface representing the current state of the user. */
+private sealed interface UserState {
```

Comment:
I think that this interface deserves it's own file. Could this `UserState` interface be used throughout the whole project, for `ProfileState` for instance.

Reply:
I'm not sure about that. I made the `UserState` interface private to this file, since it's not used anywhere else (and I don't think it will be); it's here only to model nicely how the `AuthenticationApi` used by the `Navigation` composable is selects which stateful composable should be displayed (nothing aka. splash screen, auth screen or home).

The `ProfileState` will always be used when the user is authenticated, so a `UserState` modelling if the user is logged in won't help.

Reply:
Clear !